File: Programmation/TD7/mots-fleches.pl
#!/usr/bin/env perl

use strict;
use warnings;

my $mot ;
my @possible ;

open(LEXIQUE, '<', "lexique.txt") ;

print "Veuillez entrer un mot à compléter:\n" ;
chomp($mot = <STDIN>) ;

if ($mot =~ /\./)
{
    chomp(@possible = grep(/^$mot$/, <LEXIQUE>)) ;
    
    if (@possible) # check if we have something to return ie. array not empty
    {
	print "Les possibilités sont les suivantes :\n\"" . join("\", \"", @possible) . "\"\n" ;
    }
    else
    {
	print "Aucun possibilité n'a été trouvée !\n";
    }
}
else
{
    print "Votre mot est déjà complet !\n";
    print "Veuillez remplacer les caractères inconnus par un point svp.\n" ;
}


close(LEXIQUE) ;

File: Programmation/TD7/mots-fleches.rb
#!/usr/bin/env ruby

# So chomp is not a valid method for array ?
class Array
  def chomp!
    each do |item|
      item.chomp! if item.class == String
    end
  end
end

puts "Type the word you want to complete :"
word = gets.chomp

if word =~ /\./

  possible = open('lexique.txt').grep(/\b#{word}\b/).chomp!
  
  if possible.any? # check array not empty ie. we have results
    puts "The following words would match :"
    puts "\"" + possible.join("\", \"") + "\""
  else
    puts "Sorry, no match found"
  end
else
  puts "\nYour word is already complete !!"
  puts "Please enter a dot \".\" for the unknown chars"
end

